### PR TITLE
operator sdk mig

### PR DESCRIPTION
**Pull request by odra**
~Docs and CI jobs need to be updated but the code itself can be tested an reviewed.~

## Proposed Changes

* SDK version migration to 1.x
* Docs updates (mostly links and commands);
* Tests and CI playbooks changes due to SDK changes;

## Verification Steps

* From `mbox-operator` run `molecule test -s test-local`;
* Start minikube or openshift 4 (CRC) and run `molecule test`;

## Additional Notes

Lots of changes due to SDK architectural changes.

The roles themselves had minimum to no changes, biggest changes were around tests and config files structure/organization. 

Running tests now require the `kind` CLI tool: https://github.com/kubernetes-sigs/kind

### CODE REVIEW

**Comment by morucci** [comment]
recheck

**Comment by Zlopez** [comment]
Let's merge it if the tests are passing. I will try to update the Vagrant to F33 with this already in.

File: ci/roles/openshift-osdk/defaults/main.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
-minikube_version: 1.7.3-0
-operator_sdk_version: 0.16.0
+minikube_version: 'latest'
```
Comment (Zlopez) [inline ci/roles/openshift-osdk/defaults/main.yml:1]: Not sure if this is a good idea:
+ On one side we will be notified of any error soon as it occures
- We can have errors where we don't expect them

I would be rather to use specific version, the latest is now `1.15.2`

File: ci/test-cluster.yml [hunk 2]
```diff
@@ -11,8 +11,30 @@
...
+        chdir: /usr/bin
+      become: true
     - name: Run molecule test-cluster
-      command: chdir={{ansible_user_dir}}/{{zuul.project.src_dir}}/mbox-operator molecule --debug test -s test-cluster
+      command: molecule test
+      args:
+        chdir: "{{ansible_user_dir}}/{{zuul.project.src_dir}}/mbox-operator"
       become: true
+      environment:
+        OPERATOR_IMAGE: "quay.io/lrossett/mbox-operator:dev"
```
Comment (Zlopez) [inline ci/test-cluster.yml:34]: Let's use the fedora namespace for this.

**Review by Zlopez — CHANGES_REQUESTED** [review]
Two changes remain.

**Comment by odra** [comment]
recheck

**Comment by odra** [comment]
recheck

**Review by Zlopez — APPROVED** [review]

**Comment by odra** [comment]
recheck

**Comment by Zlopez** [comment]
The [kind issue](https://github.com/bsycorp/kind/issues/39) is now fixed, you can try it with `latest-1.19`.

**Comment by odra** [comment]
latest-1.19 gave an error in my laptop as well, did it work for you?

**Comment by Zlopez** [comment]
The API wasn't available in vagrant. So I will try 1.18

**Comment by odra** [comment]
I think that route (/kubernetes-ready) doesn't work anymore, let me try a change in the tests.

**Comment by Zlopez** [comment]
In 1.18 I'm getting "Failed to establish a new connection" when creating a namespace.

**Comment by odra** [comment]
yeah I think I got it.. gimme 30 mins or so

**Comment by odra** [comment]
fingers crossed!

**Comment by Zlopez** [comment]
You got the same error I'm getting in vagrant.

**Comment by odra** [comment]
it is ignoring the KUBECONFIG env var for some reason

**Comment by Zlopez** [comment]
I noticed `server: https://localhost:10443` instead of `https://localhost:8443` you are checking in next step.

**Comment by Zlopez** [comment]
I noticed the `k8s` is trying to check the version on `http://localhost:80/version`. Checking the docker container, I don't see anything running on port 80

**Comment by odra** [comment]
it should be using the config file form KUBECONFIG env var ( /tmp/molecule/kind-test-local/kubeconfig) which has the correct url and port.

**Comment by Zlopez** [comment]
It is possible, that the k8s is looking for the file in remote machine?

**Comment by odra** [comment]
it runs on localhost, I ssh'ed into the machine and ran mocule as sudo and got the same error.

**Comment by odra** [comment]
and.. the same error happens without sudo for some reason :)

**Pull request merged**